gatherers/manifest.js

```
@@ -54,34 +38,87 @@ class Manifest extends Gather {
-      .then(node => {
-        if (!node) {
-          return this._errorManifest('No <link rel="manifest"> found in DOM.');
+
+    const manifestLoad = Manifest._resolveOnManifest();
+
+    return driver.beginNetworkCollect(manifestLoad.listener)
+      // Trigger a fetch of the manifest by Chrome.
+      .then(_ => driver.sendCommand('Page.requestAppBanner'))
+
+      // Wait on grabbing the finished manifest network request (or timing out).
+      .then(_ => manifestLoad.promise)
+
+      .then(manifestRequest => {
+        if (!manifestRequest) {
+          return Manifest._errorManifest('Timed out waiting for manifest to load.');
+        }
+        if (manifestRequest.statusCode >= 400) {
+          const reason = `${manifestRequest.statusCode}: ${manifestRequest.statusText}`;
+          return Manifest._errorManifest(
+              `Unable to fetch manifest at '${manifestRequest.url}' (${reason}).`);
         }
 
-        return node.getAttribute('href').then(manifestURL => {
-          if (!manifestURL) {
-            return this._errorManifest('No href found on <link rel="manifest">.');
-          }
-
-          return Manifest._loadFromURL(options, manifestURL)
-            .then(manifestContent => {
-              return {
-                manifest: manifestParser(manifestContent)
-              };
-            })
-            .catch(reason => {
-              return this._errorManifest(`Unable to fetch manifest at ${manifestURL}: ${reason}.`);
-            });
-        });
+        // On successful request, grab the manifest request body for parsing.
+        return driver.sendCommand('Network.getResponseBody', {
+          requestId: manifestRequest.requestId
+        })
+          .then(response => {
+            return {
+              manifest: manifestParser(response.body)
+            };
+          })
+          .catch(_ => {
+            return Manifest._errorManifest('Network recorder unable to find request for ' +
```

eating the actual error here is a tiny tiny bit scary